### ISSUE TITLE
Add warnings based on bundle size

### ISSUE THREAD

**Issue by matthewp** [stealjs/steal-tools#561]
One takeaway from the [performance audit] is that steal-tools doesn't do anything to prevent oversized bundles. It's still incredibly easy to have large bundles.

steal-tools could warn about this and link to documentation on using progressive loading. I would propose that **100k** be the maximum bundle size allowed.

We'll be super annoying about it, and require a specific flag to turn this warning off.

**Comment by justinbmeyer** [comment]
What happens if someone goes over? Would you split? Are you sure that would help?

The 100k rule seems arbitrary, especially for people not making mobile web sites.

**Comment by matthewp** [comment]
No, we wouldn't split. Splitting wouldn't necessarily help. It might a little, but if you have 800k of JavaScript that's all statically dependent on each other, that's still going to be executed as 1 chunk.  We would show a warning message with instructions (probably linking to documentation) on how to do progressive loading. WebPack is working on something similar [here](https://github.com/webpack/webpack/issues/3216), you can see a screenshot of what theres will look like.

100k is arbitrary for sure, open to a different number. Mobile traffic > 50% so I think its safe to assume everyone's targeting mobile.

**Comment by justinbmeyer** [comment]
Mobile traffic is 50% on the "open web".  The deep web, which account for probably 10x more apps is a whole other thing.

**Comment by justinbmeyer** [comment]
Sorry, I miss read the original issue.  I thought you were intending to _NOT_ produce a build if people were over.  Yeah, a warning is totally fine. 

For this to work for donejs, we probably need some lazy way of loading or setting up a `<can-datepicker>`.  Otherwise, it might be pretty hard to get the size down.
